src/Options.py

```diff
@@ -1,21 +1,84 @@
 from Options import FreeText, NumericOption, Toggle, DefaultOnToggle, Choice, TextChoice, Range, NamedRange, PerGameCommonOptions, DeathLink
 from dataclasses import make_dataclass
 from .hooks.Options import before_options_defined, after_options_defined
-from .Data import category_table, game_table
+from .Data import category_table, game_table, option_table
+from .Helpers import convertToLongString
+
 from .Locations import victory_names
 from .Items import item_table
+import logging
 
 
 class FillerTrapPercent(Range):
     """How many fillers will be replaced with traps. 0 means no additional traps, 100 means all fillers are traps."""
     range_end = 100
 
+
+def createChoiceOptions(values: dict, aliases: dict) -> dict:
+    values = {'option_' + i: v for i, v in values.items()}
+    aliases = {'alias_' + i: v for i, v in aliases.items()}
+    return {**values, **aliases}
+
 manual_options = before_options_defined({})
+manual_goal_override = {}
+for option_name, option in option_table.get('data', {}).items():
+    if option_name.startswith('_'): #To allow commenting out options
+        continue
+
+    if option_name in ['goal', 'filler_traps']:
+        if manual_options.get('goal'):
+            logging.warn("Existing Goal option found created via Hooks, it will be overwritten by Manual's generated Goal option.\nIf you want to support old yaml you will need to add alias in after_options_defined")
+        # todo do something for those situations, maybe convert, maybe warn idk for now
+        if option_name == 'goal':
+            if option['type'] != 'Choice':
+                raise Exception("a 'goal' option must be of type 'Choice'")
+            manual_goal_override['args'] = createChoiceOptions({}, option.get('aliases', {}))
+            manual_goal_override['args']['default'] = args['default'] = option.get('default', 0)
+            manual_goal_override['args']['display_name'] = option.get('display_name', option_name)
+            manual_goal_override['description'] = convertToLongString(option.get('description', ''))
+        continue
+
+    option_type = Toggle # ! I think there might be a better way to convert option['type'] -> type but I cant find it right now
```

Comment:
if someone find me a better way for this ill be able to simplify the code quite a bit

Reply:
found "locate" currently testing if that work